src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,269 @@
+                        }
+                    }
+                }
+            }
+        }
+    },
+    templates: {
+        initial: "editprefset-prefSettingAdjuster-widget"
+    },
+    invokers: {
+        saveUpdateValue: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.saveUpdateValue",
+            args: ["{that}", "{editPrefs}"]
+        },
+        cancelEditing: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.cancelEditing",
+            args: ["{that}"]
+        },
+        watchInputKeys: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.watchInputKeys",
+            args: ["{that}", "{arguments}.0"]
+        },
+        updateBlankDisabling: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBlankDisabling",
+            args: ["{that}"]
+        }
+    },
+    markupEventBindings: {
+        okButton: {
+            method: "click",
+            args: "{that}.saveUpdateValue"
+        },
+        cancelButton: {
+            method: "click",
+            args: "{that}.cancelEditing"
+        },
+        valueInput: {
+            method: "keypress",
+            args: "{that}.watchInputKeys"
+        }
+    },
+    listeners: {
+        onMarkupRendered: [{
+            funcName: "fluid.focus",
+            args: ["{that}.dom.valueInput"]
+        },
+        {
+            func: "{that}.updateBlankDisabling"
+        },
+        {
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }]
+    },
+    modelListeners: {
+        "current.blank": {
+            func: "{that}.updateBlankDisabling"
+        },
+        "current.value": [{
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }]
+    }
+});
+
+gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel = function (that) {
+    // if (that.model.metadata.schema.type === "boolean") {
+    jQuery("#pmt-boolean-value").css("display","inline");
+    jQuery("#pmt-boolean-value").css("color","black");
+    if (that.model.current.value === true)  {
+        jQuery("#pmt-boolean-value").html("true");
+    }
+    else if (that.model.current.value === false) {
+        jQuery("#pmt-boolean-value").html("false");
+    }
+};
+
+/**
+ * Updates the UI based on whether Blank is selected.
+ * If blank is selected, then the rest of the editing
+ * apparatus is disabled.
+ */
+gpii.devpmt.prefSettingAdjuster.updateBlankDisabling = function (that) {
+    var disable = that.model.current.blank;
+    that.dom.locate("valueInput").prop("disabled", disable);
+};
+
+/**
+ * watchInputKeys - Watches value input and saves
+ * on Enter.
+ *
+ * @param (Object) that - prefSettingAdjuster
+ * @param (Object) e - jQuery Event
+ */
+gpii.devpmt.prefSettingAdjuster.watchInputKeys = function (that, e) {
+    // Save the value when the enter key is pressed
+    if (e.keyCode === 13) {
+        that.saveUpdateValue();
```

Comment:
As mentioned above, pass in the save function rather than `that`.

Reply:
This is scoped better, and also fixed up a bit as it wasn't working in certain scenerios.